### PR TITLE
Fixes for CHOP 

### PR DESCRIPTION
**Pull request by GeoffNN**
- Added data normalizing with StandardScaler
- Modified the loss function to exactly match the loss function in `objective.py`

This doesn't fix the issue mentioned [here](https://github.com/benchopt/results/pull/12#issuecomment-808090858).

### CODE REVIEW

File: solvers/chop.py [hunk 1]
```diff
@@ -59,8 +61,11 @@ def set_objective(self, X, y, lmbd):
 
         device = torch.device(self.device)
 
-        self.X = torch.tensor(X).to(device)
-        self.y = torch.tensor(y > 0, dtype=torch.float64).to(device)
+        scaler = StandardScaler()
+        X = scaler.fit_transform(X)
```
Comment (agramfort) [inline solvers/chop.py:65]: this should not be done in the solver but in the dataset as it changes the problem

Comment (GeoffNN) [reply solvers/chop.py:65]: Can it be an option in the dataset?

Comment (agramfort) [reply solvers/chop.py:65]: the libsvmdata package does some preprocessing like this. I would just offer the good dataset (normalized or not).

Comment (GeoffNN) [reply solvers/chop.py:65]: Right, this was the cause of the infinity values... (Since the scaler wasn't applied at eval time) Modifying.

Comment (agramfort) [reply solvers/chop.py:65]: @GeoffNN dont change the solver but the dataset. Otherwise solvers will not be comparable

Comment (GeoffNN) [reply solvers/chop.py]: Done; I added standardization as a parameter in Covtype, so it runs on both version of the dataset. We can also just keep the standardized version if you prefer.

File: solvers/chop.py [hunk 2]
```diff
@@ -13,16 +13,17 @@ class Solver(BaseSolver):
     name = 'chop'
 
     install_cmd = 'conda'
-    requirements = ['pip:https://github.com/openopt/chop/archive/master.zip']
+    requirements = ['pip:https://github.com/openopt/chop/archive/master.zip',
+                    'pip:scikit-learn']
```
Comment (agramfort) [inline solvers/chop.py:17]:
```suggestion
    requirements = ['pip:https://github.com/openopt/chop/archive/master.zip']
```

File: datasets/covtype.py [hunk 3]
```diff
@@ -11,9 +12,18 @@ class Dataset(BaseDataset):
...
+    }
+
     def get_data(self):
         X, y = fetch_covtype(return_X_y=True)
         y[y != 2] = -1
         y[y == 2] = 1  # try to separate class 2 from the other 6 classes.
+
+        if self.standardized:
+            scaler = StandardScaler()
+            X = scaler.fit_transform(X)
```
Comment (agramfort) [inline datasets/covtype.py:26]: I would simplify and always standardize.

**Review by agramfort — COMMENTED** [review]
cool !

do you have any new figure you can share?

thx a lot @GeoffNN

File: datasets/covtype.py (hunk 3 above)
Comment (GeoffNN) [reply datasets/covtype.py:26]: Ok -- only for Covtype? It would make sense to do it for all non-sparse datasets, right?

**Comment by GeoffNN** [comment]
I'm still working on the stochastic variant -- I'll share the figures for both ASAP

**Comment by agramfort** [comment]
I don’t know about Madelon but the others should be already scaled
